pkg/apis/v1alpha1/machine.go

```diff
@@ -0,0 +1,83 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"github.com/samber/lo"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/aws/karpenter-core/pkg/apis/v1alpha5"
+	"github.com/aws/karpenter-core/pkg/scheduling"
+)
+
+type MachineSpec struct {
+	// Taints will be applied to the machine's node.
+	// +optional
+	Taints []v1.Taint `json:"taints,omitempty"`
+	// StartupTaints are taints that are applied to nodes upon startup which are expected to be removed automatically
+	// within a short period of time, typically by a DaemonSet that tolerates the taint. These are commonly used by
+	// daemonsets to allow initialization and enforce startup ordering.  StartupTaints are ignored for provisioning
+	// purposes in that pods are not required to tolerate a StartupTaint in order to have nodes provisioned for them.
+	// +optional
+	StartupTaints []v1.Taint `json:"startupTaints,omitempty"`
+	// Requirements are layered with Labels and applied to every node.
+	Requirements []v1.NodeSelectorRequirement `json:"requirements,omitempty"`
+	// Kubelet are options passed to the kubelet when provisioning nodes
+	// +optional
+	Kubelet *v1alpha5.KubeletConfiguration `json:"kubelet,omitempty"`
+	// NodeTemplateRef is a reference to an object that defines provider specific configuration
+	NodeTemplateRef *v1alpha5.ProviderRef `json:"nodeTemplateRef,omitempty"`
```

Comment:
I'm now wondering if NodeTemplate is even the right name. AWSMachineTemplate?
https://cluster-api-aws.sigs.k8s.io/topics/images/custom-amis.html
Odear.

```suggestion
	MachineTemplateRef *v1alpha1.ObjectReference `json:"nodeTemplateRef,omitempty"`
```